app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/BeaconViewModel.kt

```diff
@@ -0,0 +1,25 @@
+package ch.epfl.cs311.wanderwave.viewmodel
+
+import androidx.lifecycle.ViewModel
+import ch.epfl.cs311.wanderwave.model.data.Beacon
+import ch.epfl.cs311.wanderwave.model.data.Location
+import ch.epfl.cs311.wanderwave.model.remote.BeaconConnection
+import ch.epfl.cs311.wanderwave.model.repository.ProfileRepositoryImpl
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+@HiltViewModel
+class BeaconViewModel @Inject constructor(private val repository: ProfileRepositoryImpl) :
```

Comment:
There is a local repo and a the connection to the database, so I think it's quite good to have a view model for this to prevent injection of the different parts modules everytime, it's the same for the tracks. It's how we imagined it.

Reply:
As far as I know, it's best practice to have one viewmodel per screen. If we need data in multiple places, we can create a repository for it which combines all sources necessery and reuse that one.